RecoCTPPS/TotemRPLocal/src/TotemTimingConversions.cc

```diff
@@ -3,119 +3,157 @@
  * This is a part of CTPPS offline software.
  * Authors:
  *   Laurent Forthomme (laurent.forthomme@cern.ch)
- *   Nicola Minafra
+ *   Nicola Minafra (nicola.minafra@cern.ch)
+ *   Filip Dej
  *
  ****************************************************************************/
 
 #include "RecoCTPPS/TotemRPLocal/interface/TotemTimingConversions.h"
+#include <boost/exception/exception.hpp>
 
 //----------------------------------------------------------------------------------------------------
 
-const float TotemTimingConversions::SAMPIC_SAMPLING_PERIOD_NS = 1. / 7.8;
+const float TotemTimingConversions::SAMPIC_SAMPLING_PERIOD_NS = 1. / 7.695;
 const float TotemTimingConversions::SAMPIC_ADC_V = 1. / 256;
 const int TotemTimingConversions::SAMPIC_MAX_NUMBER_OF_SAMPLES = 64;
 const int TotemTimingConversions::SAMPIC_DEFAULT_OFFSET = 30;
+const int TotemTimingConversions::ACCEPTED_TIME_RADIUS = 4;
 
 //----------------------------------------------------------------------------------------------------
 
-TotemTimingConversions::TotemTimingConversions()
-    : calibrationFileOk_(false), calibrationFile_("/dev/null") {}
+TotemTimingConversions::TotemTimingConversions(bool mergeTimePeaks, const std::string& calibrationFile) :
+  calibrationFile_(calibrationFile),
+  calibrationFileOpened_(false),
+  mergeTimePeaks_(mergeTimePeaks)
+{}
 
 //----------------------------------------------------------------------------------------------------
 
-TotemTimingConversions::TotemTimingConversions(const std::string& calibrationFile)
-    : calibrationFileOk_(false), calibrationFile_(calibrationFile) {}
-
-//----------------------------------------------------------------------------------------------------
-
-void TotemTimingConversions::openCalibrationFile(const std::string& calibrationFile)
+void
+TotemTimingConversions::openCalibrationFile(const std::string& calibrationFile)
 {
-  if (calibrationFile!="/dev/null") // if given, the filename here overwrites the filename in the ctor
-  {
+  if (!calibrationFile.empty())
     calibrationFile_ = calibrationFile;
-  }
-
-  if (calibrationFile_!="/dev/null")
-  {
-    // Open File TODO
-    calibrationFileOk_ = true;
-  }
-
+  if (calibrationFile_.empty())
+    return;
+
+  try {
+    parsedData_.parseFile(calibrationFile_);
+  } catch (const boost::exception& e) { return; }
```

Comment:
try-catch is normally not allowed.
If the input calibration should be ignored, the job should be configured to ignore/not use a calibration.
Otherwise, this is an exception that should terminate the program.

Reply:
Indeed! The logic has now been changed to load calibration file at constructor if specified.